android/runtime/common/src/js/module.js

```diff
@@ -551,12 +551,13 @@ Module.prototype._runScript = function (source, filename) {
 	}
 	require.main = Module.main;
 
-	// This "first time" run is really only for app.js, AFAICT, and needs
-	// an activity. If app was restarted for Service only, we don't want
-	// to go this route. So added currentActivity check. (bill)
-	if (self.id === '.' && self.context.currentActivity) {
+	// Make above require() function global, if not done already.
+	if (!global.require) {
 		global.require = require;
+	}
 
+	// If this is the first script we're running, then initialize the following.
+	if (self.id === '.') {
```

Comment:
I see you removed the check to see if we're an activity or service. The toggled behavior here is about two things:
- trying to hook the debugger to pause on app start
- running the main entry point in the global context without the typical wrapping function does for every other JS file.

I'm not sure if the first would work in say a service entry point JS file?

As to the latter, we've been warning devs for a number of versions that the app.js global scope stuff would go away. so I don't know that it matters here anymore if we execute in the global context or in an injected wrapper function. (in other words, I think that's probably fine)

Reply:
With the ways things are now, we'll always launch from "ti.main" first. Even if a JS service has been started. We always load "ti.main" from `Application.onCreate()` before anything else.

If backgrounding is disabled, we really shouldn't allow a JS-service to be loaded in the background because launching and backing-out of the UI would terminate the JS runtime and kill all JS-services. It was already a broken concept before I touched the code.

I suppose I could add a safe-guard here and add an if-check for "ti.main" before hooking up the debugger?